### DISCUSSION TITLE
Assistance Needed: Dynamically Changing Image Sizes via JavaScript in Admin (Backoffice) UI

### DISCUSSION THREAD

**Discussion by aimanskie** [exoplatform/exo-community#55] [Q&A]
Hello eXo Community,

I am currently facing a challenge with dynamically adjusting image sizes in the admin UI of our eXo platform. The goal is to change the thumbnails of images from a smaller size (250x250) to a larger one (750x750) to avoid pixelation when the images are enlarged via CSS.

Here is the scenario: The thumbnails are displayed with a URL structure like /portal/rest/thumbnailImage/custom/250x250/collaboration/[image_id]. I attempted to use JavaScript to change the '250x250' segment of the URL to '750x750'. As i understood from a previous post, script is designed to be injected on every page.

The initial approach was to use querySelector and then querySelectorAll to select the relevant image elements and modify their src attribute. However, this did not seem to work as expected. The images remain pixelated, indicating that the size change might not be taking effect.

Here is the JavaScript code I used:
```
const imgElements = document.querySelectorAll('.activity-attachment>.activity-attachment-thumbnail>img.ma-auto')

imgElements.forEach((imgElement) => {
  let currentSrc = imgElement.getAttribute('src')
  let srcArray = currentSrc.split('/')
  if (srcArray.length > 5 && srcArray[5] === '250x250') {
    srcArray[5] = '750x750'
  }
  let newSrc = srcArray.join('/')
  imgElement.setAttribute('src', newSrc)
})
```
My first question would be, Is there a recommended way to dynamically adjust image sizes in the Backoffice using JavaScript?
and secondly, i've tried a few scripts but i just can't successfully querySelect any of the elements, what is wrong with how i showed above?

Any guidance or insights from the community would be greatly appreciated. Thank you in advance for your help!

Thanks!
Aiman

**Comment by rdenarie** [comment]
Hello @aimanskie 

What is the size of the original shared image ?
In fact, if the size of the image is 300x300, and you request it in 750x750, the served size will be 300x300. Image will not be upscaled. 

> Is there a recommended way to dynamically adjust image sizes in the Backoffice using JavaScript

I would say it is not recommended to do like this, because it will imply a lot of compute time and request time. Images are already downloaded by the browser in size 250x250, which take time in function of the number of images. And in your case, you add another time to change all images size, so at least you double the time for display images.

> i've tried a few scripts but i just can't successfully querySelect any of the elements, what is wrong with how i showed above?

In fact, your script executes during the page loading, and as we use VueJs framework, some images could be not loaded at all when your script runs. If you want to take care of all images, you should execute it only when the page is ready.

> Any guidance or insights from the community would be greatly appreciated. Thank you in advance for your help!

Why would you increase the image size in activity ? What is the functionnal need ?

**Reply by aimanskie** [reply]
Thank you for your response and for raising important considerations regarding image processing and load times.

To address your question, the original shared images are indeed larger than 300x300, hence our attempt to increase the displayed size to 750x750 for better readability. The functional need for this change is to ensure that images with text can be easily read within the activity feed, allowing users to comment directly without needing to enlarge the image. This enhances the user experience by making information more accessible as they scroll through the feed.

I appreciate the advice on the timing of script execution. We've successfully implemented a solution that waits for the page to be fully ready before making any changes, which is working as intended for now. We're taking it one step at a time and will consider performance optimizations as we go forward.

I believe my question has been sufficiently answered for the time being. Your guidance has been invaluable.
